lib/compat/wordpress-6.1/template-parts-screen.php

```diff
@@ -103,6 +103,8 @@ static function( $classes ) {
 		}
 	);
 
+	$extra_template_types = apply_filters( 'extra_template_types', array() );
```

Comment:
I had a look about the changes we need and you can revert all changes you did under `lib` folder.

You can add the new site editor setting in [this function](https://github.com/WordPress/gutenberg/blob/trunk/lib/compat/wordpress-6.2/site-editor.php#L15) (ex `$settings['extraTemplateTypes'] = apply_filters( 'extra_template_types', array() );`)

and rename the function to reflect changes in settings in general and update the comments.
Also noting that I think filters need special documentation([ex here](https://github.com/WordPress/gutenberg/blob/trunk/lib/compat/wordpress-6.1/blocks.php#L279)).

Reply:
Thanks for that suggestion! Just pushed the changes.